In [1]:
from fractions import Fraction as fr

In [2]:
# 先不考虑必须大于100的情况
a = fr(1, 7)
while a>1e-10:
    d = int(a.denominator/a.numerator)+1
    b = fr(1, d)
    a -= b
    print(b)
    if a.numerator==1 and a.denominator>100:
        print(a)
        break

1/8
1/57
1/3192


In [3]:
# 验证结果
fr(1,8)+fr(1,57)+fr(1,3192)

Fraction(1, 7)

In [4]:
result = [] # 全局变量用来记录所有的结果分数

In [5]:
def find_deno(x):
    '''编写一个函数求所有大于100的倒数分数'''
    global result
    while True:
        d = int(1/x)+1 #从下一个最接近的倒数分数开始减除
        while fr(1, d) in result: d+=1 # 如果该分数已经存在结果中，加一
        b = fr(1, d)
        x -= b # 减除该分数
        if d<=100: # 如果该分数分母小于等于100，递归求值
            find_deno(b)
        else:
            result.append(b) # 将该减除的分数加入结果中
            if x.denominator>100 and x.numerator==1 and x not in result:
                # 如果减去剩余的分数已经满足分子为1分母大于100，且未存在于结果中，将剩余分数加入结果中
                result.append(x)
                break

In [6]:
# 先算个简单的
result.clear()
find_deno(fr(1, 57))
sorted(result, key=lambda x:x.denominator)

[Fraction(1, 101),
 Fraction(1, 3307),
 Fraction(1, 3423),
 Fraction(1, 3541),
 Fraction(1, 3661),
 Fraction(1, 3783),
 Fraction(1, 3907),
 Fraction(1, 4033),
 Fraction(1, 4161),
 Fraction(1, 4291),
 Fraction(1, 4423),
 Fraction(1, 4557),
 Fraction(1, 4693),
 Fraction(1, 4831),
 Fraction(1, 4971),
 Fraction(1, 5113),
 Fraction(1, 5257),
 Fraction(1, 5403),
 Fraction(1, 5551),
 Fraction(1, 5701),
 Fraction(1, 5853),
 Fraction(1, 6007),
 Fraction(1, 6163),
 Fraction(1, 6321),
 Fraction(1, 6481),
 Fraction(1, 6643),
 Fraction(1, 6807),
 Fraction(1, 6973),
 Fraction(1, 7141),
 Fraction(1, 7311),
 Fraction(1, 7483),
 Fraction(1, 7657),
 Fraction(1, 7833),
 Fraction(1, 8011),
 Fraction(1, 8191),
 Fraction(1, 8373),
 Fraction(1, 8557),
 Fraction(1, 8743),
 Fraction(1, 8931),
 Fraction(1, 9121),
 Fraction(1, 9313),
 Fraction(1, 9507),
 Fraction(1, 9703),
 Fraction(1, 9901),
 Fraction(1, 10100),
 Fraction(1, 10932942),
 Fraction(1, 11713506),
 Fraction(1, 12535140),
 Fraction(1, 13399260),
 Fra

In [7]:
# 验证分子都是1
list(filter(lambda x: x.numerator!=1, result))

[]

In [8]:
# 验证分母无重复
len(set(result))==len(result)

True

In [9]:
# 验证求和结果
s = fr(0)
for x in result: s+=x
print(s)

1/57


In [10]:
# 开工
result.clear()
find_deno(fr(1, 7))
sorted(result, key=lambda x:x.denominator)

[Fraction(1, 101),
 Fraction(1, 102),
 Fraction(1, 103),
 Fraction(1, 104),
 Fraction(1, 111),
 Fraction(1, 133),
 Fraction(1, 157),
 Fraction(1, 183),
 Fraction(1, 211),
 Fraction(1, 241),
 Fraction(1, 273),
 Fraction(1, 307),
 Fraction(1, 343),
 Fraction(1, 381),
 Fraction(1, 421),
 Fraction(1, 463),
 Fraction(1, 507),
 Fraction(1, 553),
 Fraction(1, 601),
 Fraction(1, 651),
 Fraction(1, 703),
 Fraction(1, 757),
 Fraction(1, 813),
 Fraction(1, 871),
 Fraction(1, 931),
 Fraction(1, 993),
 Fraction(1, 1057),
 Fraction(1, 1123),
 Fraction(1, 1191),
 Fraction(1, 1261),
 Fraction(1, 1333),
 Fraction(1, 1407),
 Fraction(1, 1483),
 Fraction(1, 1561),
 Fraction(1, 1641),
 Fraction(1, 1723),
 Fraction(1, 1807),
 Fraction(1, 1893),
 Fraction(1, 1981),
 Fraction(1, 2071),
 Fraction(1, 2163),
 Fraction(1, 2257),
 Fraction(1, 2353),
 Fraction(1, 2451),
 Fraction(1, 2551),
 Fraction(1, 2600),
 Fraction(1, 2653),
 Fraction(1, 2757),
 Fraction(1, 2863),
 Fraction(1, 2971),
 Fraction(1, 3081),
 Fract

In [11]:
# 组合成Latex数学格式
s = '$$'
for x in sorted(result, key=lambda x:x.denominator):
    s+='+\\frac{%d}{%d}'%(x.numerator,x.denominator)
s+='$$'
print(s)

$$+\frac{1}{101}+\frac{1}{102}+\frac{1}{103}+\frac{1}{104}+\frac{1}{111}+\frac{1}{133}+\frac{1}{157}+\frac{1}{183}+\frac{1}{211}+\frac{1}{241}+\frac{1}{273}+\frac{1}{307}+\frac{1}{343}+\frac{1}{381}+\frac{1}{421}+\frac{1}{463}+\frac{1}{507}+\frac{1}{553}+\frac{1}{601}+\frac{1}{651}+\frac{1}{703}+\frac{1}{757}+\frac{1}{813}+\frac{1}{871}+\frac{1}{931}+\frac{1}{993}+\frac{1}{1057}+\frac{1}{1123}+\frac{1}{1191}+\frac{1}{1261}+\frac{1}{1333}+\frac{1}{1407}+\frac{1}{1483}+\frac{1}{1561}+\frac{1}{1641}+\frac{1}{1723}+\frac{1}{1807}+\frac{1}{1893}+\frac{1}{1981}+\frac{1}{2071}+\frac{1}{2163}+\frac{1}{2257}+\frac{1}{2353}+\frac{1}{2451}+\frac{1}{2551}+\frac{1}{2600}+\frac{1}{2653}+\frac{1}{2757}+\frac{1}{2863}+\frac{1}{2971}+\frac{1}{3081}+\frac{1}{3193}+\frac{1}{3194}+\frac{1}{3307}+\frac{1}{3308}+\frac{1}{3423}+\frac{1}{3424}+\frac{1}{3434}+\frac{1}{3541}+\frac{1}{3542}+\frac{1}{3661}+\frac{1}{3662}+\frac{1}{3783}+\frac{1}{3784}+\frac{1}{3907}+\frac{1}{3908}+\frac{1}{4033}+\frac{1}{4034}+\fr

### 下面就是看结果的时候了
$$+\frac{1}{101}+\frac{1}{102}+\frac{1}{103}+\frac{1}{104}+\frac{1}{111}+\frac{1}{133}+\frac{1}{157}+\frac{1}{183}+\frac{1}{211}+\frac{1}{241}+\frac{1}{273}+\frac{1}{307}+\frac{1}{343}+\frac{1}{381}+\frac{1}{421}+\frac{1}{463}+\frac{1}{507}+\frac{1}{553}+\frac{1}{601}+\frac{1}{651}+\frac{1}{703}+\frac{1}{757}+\frac{1}{813}+\frac{1}{871}+\frac{1}{931}+\frac{1}{993}+\frac{1}{1057}+\frac{1}{1123}+\frac{1}{1191}+\frac{1}{1261}+\frac{1}{1333}+\frac{1}{1407}+\frac{1}{1483}+\frac{1}{1561}+\frac{1}{1641}+\frac{1}{1723}+\frac{1}{1807}+\frac{1}{1893}+\frac{1}{1981}+\frac{1}{2071}+\frac{1}{2163}+\frac{1}{2257}+\frac{1}{2353}+\frac{1}{2451}+\frac{1}{2551}+\frac{1}{2600}+\frac{1}{2653}+\frac{1}{2757}+\frac{1}{2863}+\frac{1}{2971}+\frac{1}{3081}+\frac{1}{3193}+\frac{1}{3194}+\frac{1}{3307}+\frac{1}{3308}+\frac{1}{3423}+\frac{1}{3424}+\frac{1}{3434}+\frac{1}{3541}+\frac{1}{3542}+\frac{1}{3661}+\frac{1}{3662}+\frac{1}{3783}+\frac{1}{3784}+\frac{1}{3907}+\frac{1}{3908}+\frac{1}{4033}+\frac{1}{4034}+\frac{1}{4161}+\frac{1}{4162}+\frac{1}{4291}+\frac{1}{4292}+\frac{1}{4423}+\frac{1}{4424}+\frac{1}{4557}+\frac{1}{4558}+\frac{1}{4693}+\frac{1}{4694}+\frac{1}{4831}+\frac{1}{4832}+\frac{1}{4971}+\frac{1}{4972}+\frac{1}{5100}+\frac{1}{5113}+\frac{1}{5114}+\frac{1}{5257}+\frac{1}{5258}+\frac{1}{5259}+\frac{1}{5403}+\frac{1}{5404}+\frac{1}{5405}+\frac{1}{5551}+\frac{1}{5552}+\frac{1}{5553}+\frac{1}{5701}+\frac{1}{5702}+\frac{1}{5703}+\frac{1}{5853}+\frac{1}{5854}+\frac{1}{5855}+\frac{1}{6007}+\frac{1}{6008}+\frac{1}{6009}+\frac{1}{6163}+\frac{1}{6164}+\frac{1}{6165}+\frac{1}{6321}+\frac{1}{6322}+\frac{1}{6323}+\frac{1}{6481}+\frac{1}{6482}+\frac{1}{6483}+\frac{1}{6643}+\frac{1}{6644}+\frac{1}{6645}+\frac{1}{6807}+\frac{1}{6808}+\frac{1}{6809}+\frac{1}{6973}+\frac{1}{6974}+\frac{1}{6975}+\frac{1}{7141}+\frac{1}{7142}+\frac{1}{7143}+\frac{1}{7311}+\frac{1}{7312}+\frac{1}{7313}+\frac{1}{7483}+\frac{1}{7484}+\frac{1}{7485}+\frac{1}{7657}+\frac{1}{7658}+\frac{1}{7659}+\frac{1}{7833}+\frac{1}{7834}+\frac{1}{7835}+\frac{1}{8011}+\frac{1}{8012}+\frac{1}{8013}+\frac{1}{8191}+\frac{1}{8192}+\frac{1}{8193}+\frac{1}{8194}+\frac{1}{8373}+\frac{1}{8374}+\frac{1}{8375}+\frac{1}{8376}+\frac{1}{8557}+\frac{1}{8558}+\frac{1}{8559}+\frac{1}{8560}+\frac{1}{8743}+\frac{1}{8744}+\frac{1}{8745}+\frac{1}{8746}+\frac{1}{8931}+\frac{1}{8932}+\frac{1}{8933}+\frac{1}{8934}+\frac{1}{9121}+\frac{1}{9122}+\frac{1}{9123}+\frac{1}{9124}+\frac{1}{9313}+\frac{1}{9314}+\frac{1}{9315}+\frac{1}{9316}+\frac{1}{9507}+\frac{1}{9508}+\frac{1}{9509}+\frac{1}{9510}+\frac{1}{9703}+\frac{1}{9704}+\frac{1}{9705}+\frac{1}{9706}+\frac{1}{9901}+\frac{1}{9902}+\frac{1}{9903}+\frac{1}{9904}+\frac{1}{10100}+\frac{1}{12210}+\frac{1}{17556}+\frac{1}{24492}+\frac{1}{33306}+\frac{1}{44310}+\frac{1}{57840}+\frac{1}{74256}+\frac{1}{93942}+\frac{1}{117306}+\frac{1}{144780}+\frac{1}{176820}+\frac{1}{213906}+\frac{1}{256542}+\frac{1}{305256}+\frac{1}{360600}+\frac{1}{423150}+\frac{1}{493506}+\frac{1}{572292}+\frac{1}{660156}+\frac{1}{757770}+\frac{1}{865830}+\frac{1}{985056}+\frac{1}{1116192}+\frac{1}{1260006}+\frac{1}{1417290}+\frac{1}{1588860}+\frac{1}{1775556}+\frac{1}{1978242}+\frac{1}{2197806}+\frac{1}{2435160}+\frac{1}{2691240}+\frac{1}{2967006}+\frac{1}{3263442}+\frac{1}{3581556}+\frac{1}{3922380}+\frac{1}{4286970}+\frac{1}{4676406}+\frac{1}{5091792}+\frac{1}{5097624}+\frac{1}{5468124}+\frac{1}{5534256}+\frac{1}{5858464}+\frac{1}{6004950}+\frac{1}{6269340}+\frac{1}{6505050}+\frac{1}{6701460}+\frac{1}{7035756}+\frac{1}{7155544}+\frac{1}{7598292}+\frac{1}{7632324}+\frac{1}{8132544}+\frac{1}{8193906}+\frac{1}{8656960}+\frac{1}{8823870}+\frac{1}{9206340}+\frac{1}{9213768}+\frac{1}{9489480}+\frac{1}{9732604}+\frac{1}{9781464}+\frac{1}{10192056}+\frac{1}{10273050}+\frac{1}{10383124}+\frac{1}{10835700}+\frac{1}{10932942}+\frac{1}{11012124}+\frac{1}{11421154}+\frac{1}{11669280}+\frac{1}{11713506}+\frac{1}{12030018}+\frac{1}{12355420}+\frac{1}{12535140}+\frac{1}{12662910}+\frac{1}{13071384}+\frac{1}{13320454}+\frac{1}{13399260}+\frac{1}{13818024}+\frac{1}{14003280}+\frac{1}{14307306}+\frac{1}{14596204}+\frac{1}{14712030}+\frac{1}{15260742}+\frac{1}{15406800}+\frac{1}{15447352}+\frac{1}{16209900}+\frac{1}{16250700}+\frac{1}{16261056}+\frac{1}{16777215}+\frac{1}{17000340}+\frac{1}{17128804}+\frac{1}{17309760}+\frac{1}{17530968}+\frac{1}{17685100}+\frac{1}{17819344}+\frac{1}{18042024}+\frac{1}{18309840}+\frac{1}{18408390}+\frac{1}{18667590}+\frac{1}{18991284}+\frac{1}{19114383}+\frac{1}{19545768}+\frac{1}{19558506}+\frac{1}{19945155}+\frac{1}{19977520}+\frac{1}{20454574}+\frac{1}{20761692}+\frac{1}{20802720}+\frac{1}{21001680}+\frac{1}{21394710}+\frac{1}{21687648}+\frac{1}{22019556}+\frac{1}{22064724}+\frac{1}{22366890}+\frac{1}{22600515}+\frac{1}{23167624}+\frac{1}{23333730}+\frac{1}{23371834}+\frac{1}{23541903}+\frac{1}{24311364}+\frac{1}{24410268}+\frac{1}{24512400}+\frac{1}{24705870}+\frac{1}{25482930}+\frac{1}{25496940}+\frac{1}{26137656}+\frac{1}{26590564}+\frac{1}{26725360}+\frac{1}{27630792}+\frac{1}{27733920}+\frac{1}{27997644}+\frac{1}{28913760}+\frac{1}{29187006}+\frac{1}{29314824}+\frac{1}{30130852}+\frac{1}{30677944}+\frac{1}{30808050}+\frac{1}{31385970}+\frac{1}{32088060}+\frac{1}{32495700}+\frac{1}{32679900}+\frac{1}{33546240}+\frac{1}{34251756}+\frac{1}{35053564}+\frac{1}{36078042}+\frac{1}{36611124}+\frac{1}{37976406}+\frac{1}{38220024}+\frac{1}{39881380}+\frac{1}{39948720}+\frac{1}{41596320}+\frac{1}{41996880}+\frac{1}{43365984}+\frac{1}{44122806}+\frac{1}{45191524}+\frac{1}{46328442}+\frac{1}{47074104}+\frac{1}{48615756}+\frac{1}{49014900}+\frac{1}{50986740}+\frac{1}{53443410}+\frac{1}{55987806}+\frac{1}{58621992}+\frac{1}{61348056}+\frac{1}{64168110}+\frac{1}{67084290}+\frac{1}{70098756}+\frac{1}{73213692}+\frac{1}{76431306}+\frac{1}{79753830}+\frac{1}{83183520}+\frac{1}{86722656}+\frac{1}{90373542}+\frac{1}{94138506}+\frac{1}{98019900}+\frac{1}{142085361198620}+\frac{1}{195664126616420}+\frac{1}{266151728828720}+\frac{1}{357931010270504}+\frac{1}{476293513066160}+\frac{1}{627584375827640}+\frac{1}{819363913413500}+\frac{1}{1060587114166580}+\frac{1}{1361802333238004}$$